src/scripts/main.js

```diff
@@ -1,3 +1,24 @@
 'use strict';
 
-// write your code here
+function cloneColumn(tableElement, columnIndex) {
+  const headRows = tableElement.querySelectorAll('thead tr');
+  const bodyRows = tableElement.querySelectorAll('tbody tr');
+  const footRows = tableElement.querySelectorAll('tfoot tr');
+
+  function insertClonedCell(rows) {
+    rows.forEach((row) => {
+      const cells = row.children;
+      const clonedCell = cells[columnIndex].cloneNode(true);
+
+      row.insertBefore(clonedCell, cells[cells.length - 1]);
```

Comment:
The `insertBefore` method is used here to insert the cloned cell before the last cell in the row. If the intention is to clone the column at the same index, you should use `insertBefore(clonedCell, cells[columnIndex + 1])` to ensure the cloned cell is inserted immediately after the original cell at `columnIndex`.